wfe/wfe.go

```diff
@@ -280,25 +360,41 @@ func (wfe *WebFrontEndImpl) handleKey(
 func (wfe *WebFrontEndImpl) Handler() http.Handler {
 	m := http.NewServeMux()
 	// GET only handlers
-	wfe.HandleFunc(m, DirectoryPath, wfe.Directory, "GET")
+	wfe.HandleFunc(m, DirectoryPath, wfe.Directory, false, "GET")
 	// Note for noncePath: "GET" also implies "HEAD"
-	wfe.HandleFunc(m, noncePath, wfe.Nonce, "GET")
-	wfe.HandleFunc(m, RootCertPath, wfe.handleCert(wfe.ca.GetRootCert()), "GET")
-	wfe.HandleFunc(m, rootKeyPath, wfe.handleKey(wfe.ca.GetRootKey()), "GET")
-	wfe.HandleFunc(m, intermediateCertPath, wfe.handleCert(wfe.ca.GetIntermediateCert()), "GET")
-	wfe.HandleFunc(m, intermediateKeyPath, wfe.handleKey(wfe.ca.GetIntermediateKey()), "GET")
+	wfe.HandleFunc(m, noncePath, wfe.Nonce, false, "GET")
+	wfe.HandleFunc(m, RootCertPath, wfe.handleCert(
+		func(no int) *core.Certificate {
```

Comment:
You don't have to define a local function here, you can just write:

```
wfe.HandleFunc(m, RootCertPath, wfe.handleCert(wfe.ca.GetRootCert, ...
```

This is an unusual feature of Go called [method expressions](https://golang.org/ref/spec#Method_expressions).

I believe the same applies to the funcs you define in the other HandleFuncs below.

Reply:
Ah, so Go behaves similarly to Java 8+ where this is also possible. Neat. Makes it a lot easier :)

Reply:
Done.